### PR TITLE
bugfix: fix GlobalSession cannot be cleared normally when delayed deletion is enabled

### PR DESCRIPTION
**Pull request by GoodBoyCoder**
…alSession

- [x] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
GlobalSession在延迟删除的情况下，由于提前设置了Rollbacked状态导致定时任务无法即时处理。目前放弃提前设置状态，改为由DefaultCore#doGlobalRollback的执行结果来确认是否返回Rollbacked状态

In the case of delayed deletion of GlobalSession, the scheduled task cannot be processed immediately because the Rollbacked state is set in advance. At present, the advance setting state is abandoned, and the execution result of DefaultCore#doGlobalRollback is used to confirm whether to return the Rollbacked state.

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM  pr登记一下

File: changes/1.5.0.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ Seata 是一款开源的分布式事务解决方案，提供高性能和简单
   - [[#4310](https://github.com/seata/seata/pull/4310)] 修复通过"SELECT LAST_INSERT_ID"获取mysql数据库自增id失败的问题
   - [[#4331](https://github.com/seata/seata/pull/4331)] 修复使用ONLY_CARE_UPDATE_COLUMNS配置可能出现的脏写校验异常
   - [[#4408](https://github.com/seata/seata/pull/4408)] 修复容器环境中设置环境变量无效的问题
+  - [[#4438](https://github.com/seata/seata/pull/4438)] 修复GlobalSession在延迟删除的情况下无法被正常删除的问题
```
Comment (a364176773) [inline changes/1.5.0.md:100]: 修复develop版本file模式下GlobalSession在延迟删除的情况下无法被正常删除的问题
解释可以写精确点,避免有用户认为是1.4.2有这个问题

Comment (GoodBoyCoder) [reply changes/1.5.0.md]: Done

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**